### DISCUSSION TITLE
synchronize_session='fetch' ignores returning-option

### DISCUSSION THREAD

**Discussion by valq7711** [sqlalchemy/sqlalchemy#11857] [Usage Questions] [answered]
#### Delete with  `returning`-option and  `synchronize_session='fetch'`:
```python
q = (
    delete(<some_table>).where(<some_table>.id == <id>)
    .returning(<some_table>.id)
    .execution_options(synchronize_session='fetch')
)
res = await session.execute(q)
print(sqlalchemy.__version__)
print('rows:', res.all())
print('rowcount:', res.rowcount)
```

output:
```
1.4.54
rows: []
rowcount: 1
```

#### The same, but without `synchronize_session='fetch'`:
```python
q = (
    delete(<some_table>).where(<some_table>.id == <id>)
    .returning(<some_table>.id)
)
res = await session.execute(q)
print(sqlalchemy.__version__)
print('rows:', res.all())
print('rowcount:', res.rowcount)
```
output:
```
1.4.54
rows: [(UUID('8a874afd-3c2c-4316-96b3-6932d6264dbd'),)]
rowcount: 1
```

I think this is definitely a bug, not a feature. (database - PostgreSQL)

**Comment by zzzeek** [comment, marked as answer]
Hi -

this was not supported until 2.0. upgrade to 2.0 and this should be working

**Comment by valq7711** [comment]
Thanks for the quick reply!
But as far as I know, 2.0 is not fully compatible with 1.4, so upgrade needs some extra work :(.
So it's cheaper to use a workaround (given that I only need the IDs):
```python
stmt = str(q.compile(dialect=postgresql.dialect(), compile_kwargs={"literal_binds": True}))
res = await session.execute(stmt)
```

**Reply by CaselIT** [reply]
it's likely easier to expire the object instead, so you don't have to use `execution_options(synchronize_session='fetch')`

**Reply by valq7711** [reply]
oh, my example is very simple to just highlight the issue, in fact I have complex filter involved other tables and without `synchronize_session='fetch'` I get the error `Can't evaluate criteria against alternate class ...` 
The IDs of the records to be deleted are unknown.

**Reply by CaselIT** [reply]
if the object are expired from the connection synchronize should not come into play.
But in any case you can execute on the connection instead of doing that
```
conn = await session.connection()
res = await conn.execute(stmt)
```
